3.processing_features/scripts/1.annotate_sc.py

```diff
@@ -29,28 +28,45 @@
 
 # output directory for annotated data
 output_dir = pathlib.Path("./data/annotated_data")
-# if directory if doesn't exist, will not raise error if it already exists
-os.makedirs(output_dir, exist_ok=True)
+output_dir.mkdir(exist_ok=True)
+
+# directory with metadata
+metadata_dir = pathlib.Path("../0.download_data/metadata/")
 
 # load in dicionary from yaml file
 dictionary_path = pathlib.Path("./plate_info_dictionary.yaml")
 with open(dictionary_path) as file:
     plate_info_dictionary = yaml.load(file, Loader=yaml.FullLoader)
 
-# add paths to dictionary that are used for annotation
-plate_info_dictionary["Plate_1"]["platemap_path"] = str(pathlib.Path("../0.download_data/metadata/platemap_NF1_plate1.csv"))
-plate_info_dictionary["Plate_2"]["platemap_path"] = str(pathlib.Path("../0.download_data/metadata/platemap_NF1_plate2.csv"))
-# both plates 3 and 3 prime use the same platemap file (same metadata)
-plate_info_dictionary["Plate_3"]["platemap_path"] = str(pathlib.Path("../0.download_data/metadata/platemap_NF1_plate3.csv"))
-plate_info_dictionary["Plate_3_prime"]["platemap_path"] = str(pathlib.Path("../0.download_data/metadata/platemap_NF1_plate3.csv"))
+
+# ## Add metadata paths to loaded in dictionary
+
+# In[3]:
+
+
+# add path to platemaps for each plate 
+for plate, _ in plate_info_dictionary.items():
+    # since Plate_3_prime has the same platemap as Plate_3, we need an else statement so that we make sure it adds the 
+    # path that was given to Plate_3
+    if plate != "Plate_3_prime":
+        # match the naming format of the plates to the platemap file
+        plate_info_dictionary[plate]["platemap_path"] = str(
```

Comment:
Consider combining this for loop with the one below, which also iterate through `plate_info_dictionary`

Reply:
It is a good idea, but in this context, that other `for` loop is already pretty dense with all of the functions to annotate + rename + move around columns.

For interpretability, it is easier to have the two `for` loops that perform different tasks.